### PR TITLE
fix: missing where key in plugin specified templates

### LINKED ISSUE #7843: Bug: Query with where clause fails in edit mode.
## Description

Query with where clause fails in edit mode.

### Steps to reproduce the behaviour:

Note: This issue can be reproduced with pre-existing query where `key: where` is missing in `pluginSpecifiedTemplates` in `action` object.

1. Go to Datasources
2. Open a query on GoogleSheets
3.  Add a where clause
4. Click Run
5. Query will fail

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

### PR DESCRIPTION
**Pull request by rashmi-rai**
## Description

Fixes #7843 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/where-key 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.79 **(-0.02)** | 36.91 **(-0.04)** | 33.8 **(0)** | 55.4 **(-0.02)**
 :red_circle: | app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx | 20.29 **(-5.2)** | 0 **(0)** | 0 **(0)** | 21.54 **(-5.54)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx [hunk 1]
```diff
@@ -52,6 +58,25 @@ function ActionForm(props: Props) {
     dispatch(deleteAction({ id: apiId, name: actionName }));
   };
 
+  if (!!props.difference) {
```
Comment (mohanarpit) [inline app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx:61]: Can we add a comment here explaining what this for-loop is accomplishing? Basically, helping the reader understand the "why" of the code. If the comment can mention the structure/type of `props.difference` object, that'll be great to visualize the operations in the loop.

**Comment by trishaanand** [comment]
@rashmi-rai : Could you please add a test case for this?

**Comment by rashmi-rai** [comment]
> @rashmi-rai : Could you please add a test case for this?

@trishaanand sure. But can I add it in the next PR?

File: app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx [hunk 2]
```diff
@@ -103,7 +155,7 @@ const mapStateToProps = (state: AppState, props: any) => {
   }
   merge(initialValues, getConfigInitialValues(settingConfig));
   merge(initialValues, action);
-
+  const difference = diff(action, initialValues);
```
Comment (ayushpahwa) [inline app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx:158]: Can you add a comment here and also improve the var name?

Comment (rashmi-rai) [reply app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx]: Added

**Review by ayushpahwa — APPROVED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**